specification/ml/_types/inference.ts

```diff
@@ -328,3 +348,79 @@ export class FillMaskInferenceUpdateOptions {
   /** The field that is added to incoming documents to contain the inference prediction. Defaults to predicted_value. */
   results_field?: string
 }
+
+export class TrainedModelEntities {
+  class_name: string
+  class_probability: double
+  entity: string
+  start_pos: integer
+  end_pos: integer
+}
+export class TopClassEntry {
+  class_name: string
+  class_probability: double
+  class_score: double
+}
+
+export class TrainedModelInferenceClassImportance {
+  class_name: string
+  importance: double
+}
+
+export class TrainedModelInferenceFeatureImportance {
+  feature_name: string
+  importance?: double
+  classes?: TrainedModelInferenceClassImportance[]
+}
+
+export type PredictedValue = string | double | boolean | integer
+
+/** @variants container */
```

Comment:
I don't this class is intended to be a container, the "container" variant means properties are mutually exclusive: https://github.com/elastic/elasticsearch-specification/blob/main/docs/modeling-guide.md#container

Since the `Container` suffix is typically used for these cases, maybe we should use a different name here? `TrainedModelInferenceResult`?